csharp/extractor/Semmle.Extraction.CSharp.Standalone/Runtime.cs

```diff
@@ -22,7 +22,7 @@ internal partial class Runtime
 
         public Runtime(IDotNet dotNet) => this.dotNet = dotNet;
 
-        internal sealed class RuntimeVersion : IComparable<RuntimeVersion>
+        internal record RuntimeVersion : IComparable<RuntimeVersion>
```

Comment:
@michaelnebel Somewhat independently from this change. I think we're not using the `IComparable<>` anywhere in the application. We seem to rely on the order of the runtimes that the dotnet CLI is returning to us, which happens to be chronological, and therefore we're using the latest installed version.

Reply:
Could you elaborate a bit? The `AddOrUpdate` method uses the `Compare` method to decide whether it should "replace" the runtime version. I didn't feel confident that `dotnet --list-runtimes` would list the "newest" runtime first (maybe it would be based on installation order or something else).

Reply:
Ahh, sorry, I didn't realize we had our custom `AddOrUpdate`. For some reason, I was thinking about  `ConcurrentDictionary.AddOrUpdate`. Could you rename `DictionaryExtensions.AddOrUpdate` to `AddOrUpdateToLatest` to avoid confusions?

Reply:
Yes, sure 😄